src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/model/request/VideoLinkBookingCreateRequest.kt

```diff
@@ -0,0 +1,94 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.model.request
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import io.swagger.v3.oas.annotations.media.Schema
+import jakarta.validation.Valid
+import jakarta.validation.constraints.AssertTrue
+import jakarta.validation.constraints.Future
+import jakarta.validation.constraints.NotBlank
+import jakarta.validation.constraints.NotEmpty
+import jakarta.validation.constraints.NotNull
+import jakarta.validation.constraints.Size
+import java.net.URI
+import java.time.LocalDate
+import java.time.LocalTime
+
+data class VideoLinkBookingCreateRequest(
+
+  @field:NotNull(message = "The video link booking type is mandatory")
+  @Schema(description = "The booking type", example = "COURT")
+  val bookingType: BookingType?,
+
+  @field:Valid
+  @field:NotEmpty(message = "At least one prisoner must be supplied for a video link booking")
+  @Schema(description = "The prisoner or prisoners associated with the video link booking")
+  val prisoners: List<PrisonerDetails>,
+
+  @field:Size(max = 400, message = "Comments for the video link booking cannot not exceed {max} characters")
+  val comments: String?,
+)
+
+enum class BookingType {
+  COURT,
+  PROBATION,
+}
+
+data class PrisonerDetails(
+
+  @field:NotEmpty(message = "Prison code is mandatory")
+  @field:Size(max = 3, message = "Prison code should not exceed {max} characters")
+  @Schema(description = "The prison code for the prisoner", example = "PVI")
+  val prisonCode: String?,
+
+  @field:NotBlank(message = "Prisoner number is mandatory")
+  @field:Size(max = 7, message = "Prisoner number must not exceed {max} characters")
+  @Schema(description = "The prisoner number (NOMIS ID)", example = "A1234AA")
+  val prisonerNumber: String?,
+
+  @field:Valid
+  @field:NotEmpty(message = "At least one appointment must be supplied for the prisoner")
+  @Schema(description = "The appointments associated with the prisoner")
+  val appointments: List<Appointment>,
+)
+
+data class Appointment(
+  @field:NotNull(message = "The appointment type for the appointment is mandatory")
+  @Schema(description = "The appointment type", example = "TO BE DEFINED")
+  val type: AppointmentType?,
+
+  @field:NotEmpty(message = "The location identifier for the appointment is mandatory")
+  @Schema(description = "The location identifier for the appointment")
+  val locationId: String?,
```

Comment:
Should probably be limiting the number of characters for any String supplied values ....